modules/build/src/main/scala/scala/build/input/ScalaCliInvokeData.scala

```diff
@@ -0,0 +1,37 @@
+package scala.build.input
+
+/** Stores information about how the program has been evoked
+  *
+  * @param progName
+  *   the actual Scala CLI program name which was run
+  * @param subCommand
+  *   the sub-command that was invoked by user
+  */
+
+case class ScalaCliInvokeData(progName: String, subCommandName: String, subCommand: SubCommand)
+
+enum SubCommand:
+  case Default extends SubCommand
+  case Shebang extends SubCommand
+  case Other   extends SubCommand
+
+object ScalaCliInvokeData {
+
+  // Copied from https://github.com/scalacenter/bloop/blob/main/shared/src/main/scala/bloop/io/Environment.scala
```

Comment:
Better to use permalink - https://github.com/scalacenter/bloop/blob/a249e0a710ce169ca05d0606778f96f44a398680/shared/src/main/scala/bloop/io/Environment.scala